packages/@aws-cdk/aws-dynamodb/lib/replica-provider.ts

```diff
@@ -84,6 +84,14 @@ export class ReplicaProvider extends NestedStack {
       }),
     );
 
+    // Required for replica table deletion
+    this.onEventHandler.addToRolePolicy(
+      new iam.PolicyStatement({
+        actions: ['dynamodb:DeleteTable', 'dynamodb:DeleteTableReplica'],
```

Comment:
Seems like only `DeleteTableReplica` should be required, is `DeleteTable` also necessary?

Reply:
While i was running integration test from scratch, it was requesting this permission too. @MrArnoldPalmer 

>  ❌ Deployment failed: Error: Stack Deployments Failed: Error: The stack named global-replica-single-region-delete failed to deploy: CREATE_FAILED (The following resource(s) failed to create: [CustomResourceLambdauseast21CE92864]. ): Received response status [FAILED] from custom resource. Message returned: User: arn:aws:sts:::assumed-role/global-replica-single-reg-AWS679f53fac002430cb0da5-773B4IWO1P2L/global-replica-single-reg-AWS679f53fac002430cb0da5-GPv2ZsyPAzba is not authorized to perform: dynamodb:DeleteTable on resource: arn:aws:dynamodb:us-east-2::table/global-replica-region-table because no identity-based policy allows the dynamodb:DeleteTable action (RequestId: cb9c2ab7-b18b-4e5c-9687-1853379dc36a)
>     at exports.deployStacks (/Users/ahmetsoykan/projects/repos/github.com/ahmetsoykan/aws-cdk/packages/aws-cdk/lib/deploy.js:43:15)
>     at async CdkToolkit.deploy (/Users/ahmetsoykan/projects/repos/github.com/ahmetsoykan/aws-cdk/packages/aws-cdk/lib/cdk-toolkit.js:266:13)
>     at async exec (/Users/ahmetsoykan/projects/repos/github.com/ahmetsoykan/aws-cdk/packages/aws-cdk/lib/cli.js:374:16)
> 
> Stack Deployments Failed: Error: The stack named global-replica-single-region-delete failed to deploy: CREATE_FAILED (The following resource(s) failed to create: [CustomResourceLambdauseast21CE92864]. ): Received response status [FAILED] from custom resource. Message returned: User: arn:aws:sts:::assumed-role/global-replica-single-reg-AWS679f53fac002430cb0da5-773B4IWO1P2L/global-replica-single-reg-AWS679f53fac002430cb0da5-GPv2ZsyPAzba is not authorized to perform: dynamodb:DeleteTable on resource: arn:aws:dynamodb:us-east-2::table/global-replica-region-table because no identity-based policy allows the dynamodb:DeleteTable action (RequestId: cb9c2ab7-b18b-4e5c-9687-1853379dc36a)
>   FAILED     integ.global-replica-single-region-delete-global-replica-single-region-delete-integ/DefaultTest (undefined/us-east-1) 619.274s
>       Integration test failed: Error: Command exited with status 1

Reply:
I also saw that these permissions are documented at [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/V2gt_IAM.html)